controllers/components/dashboard/dashboard_controller.go

```diff
@@ -0,0 +1,408 @@
+		cluster.Unknown:          "odh-admins",
+	}[platform]
+
+	sectionTitle := map[cluster.Platform]string{
+		cluster.SelfManagedRhods: "OpenShift Self Managed Services",
+		cluster.ManagedRhods:     "OpenShift Managed Services",
+		cluster.OpenDataHub:      "OpenShift Open Data Hub",
+		cluster.Unknown:          "OpenShift Open Data Hub",
+	}[platform]
+
+	consoleLinkDomain, err := cluster.GetDomain(ctx, cli)
+	if err != nil {
+		return nil, fmt.Errorf("error getting console route URL %s : %w", consoleLinkDomain, err)
+	}
+	consoleURL := map[cluster.Platform]string{
+		cluster.SelfManagedRhods: "https://rhods-dashboard-" + dscispec.ApplicationsNamespace + "." + consoleLinkDomain,
+		cluster.ManagedRhods:     "https://rhods-dashboard-" + dscispec.ApplicationsNamespace + "." + consoleLinkDomain,
+		cluster.OpenDataHub:      "https://odh-dashboard-" + dscispec.ApplicationsNamespace + "." + consoleLinkDomain,
+		cluster.Unknown:          "https://odh-dashboard-" + dscispec.ApplicationsNamespace + "." + consoleLinkDomain,
+	}[platform]
+
+	return map[string]string{
+		"admin_groups":  adminGroups,
+		"dashboard-url": consoleURL,
+		"section-title": sectionTitle,
+	}, nil
+}
+
+// Action implementations
+
+type InitializeAction struct {
+	actions.BaseAction
+}
+
+func (a *InitializeAction) Execute(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Implement initialization logic
+	log := logf.FromContext(ctx).WithName(ComponentNameUpstream)
+
+	imageParamMap := map[string]string{
+		"odh-dashboard-image": "RELATED_IMAGE_ODH_DASHBOARD_IMAGE",
+	}
+	manifestMap := map[cluster.Platform]string{
+		cluster.SelfManagedRhods: PathDownstream + "/onprem",
+		cluster.ManagedRhods:     PathDownstream + "/addon",
+		cluster.OpenDataHub:      PathUpstream,
+		cluster.Unknown:          PathUpstream,
+	}
+	DefaultPath = manifestMap[rr.Platform]
+
+	rr.Manifests = manifestMap
+
+	if err := deploy.ApplyParams(DefaultPath, imageParamMap); err != nil {
+		log.Error(err, "failed to update image", "path", DefaultPath)
+	}
+
+	return nil
+}
+
+type SupportDevFlagsAction struct {
+	actions.BaseAction
+}
+
+func (a *SupportDevFlagsAction) Execute(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dashboard, ok := rr.Instance.(*componentsv1.Dashboard)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Dashboard)", rr.Instance)
+	}
+
+	if dashboard.Spec.DevFlags == nil {
+		return nil
+	}
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(dashboard.Spec.DevFlags.Manifests) != 0 {
+		manifestConfig := dashboard.Spec.DevFlags.Manifests[0]
+		if err := deploy.DownloadManifests(ctx, ComponentNameUpstream, manifestConfig); err != nil {
+			return err
+		}
+		if manifestConfig.SourcePath != "" {
+			rr.Manifests[rr.Platform] = filepath.Join(deploy.DefaultManifestPath, ComponentNameUpstream, manifestConfig.SourcePath)
+		}
+	}
+
+	if rr.DSCI.Spec.DevFlags != nil {
```

Comment:
Remove that, deprecated by coming https://github.com/opendatahub-io/opendatahub-operator/pull/1307